File: fox-hunt-domain/src/main/java/com/itechart/foxhunt/domain/entity/OrganizationUserRoleEntity.java
package com.itechart.foxhunt.domain.entity;

import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name = "organization_user_role", schema = "fh_admin")
public class OrganizationUserRoleEntity {

  @EmbeddedId
  private OrganizationUserRoleEntityPK id;

  @OneToOne(optional = false, cascade = CascadeType.PERSIST)
  @JoinColumn(name = "user_id", referencedColumnName = "app_user_id")
  @MapsId("userId")
  private UserEntity userEntity;

  @OneToOne(optional = false, cascade = CascadeType.PERSIST)
  @JoinColumn(name = "role_id", referencedColumnName = "role_id")
  @MapsId("roleId")
  private RoleEntity roleEntity;

  @Column(name = "is_active", nullable = false)
  private boolean isActive;
}

File: fox-hunt-api-gateway/src/common/entities/organizationUserRole.entity.ts
import { Type } from 'class-transformer';
import { Entity, Column, JoinColumn, ManyToOne, PrimaryColumn } from 'typeorm';
import { RoleEntity } from './role.entity';
import { UserEntity } from './user.entity';

@Entity('organization_user_role', { schema: 'fh_admin' })
export class OrganizationUserRoleEntity {
  @PrimaryColumn()
  role_id: number;

  @ManyToOne(() => RoleEntity)
  @JoinColumn({ name: 'role_id' })
  role: RoleEntity;

  @Column('integer', { name: 'user_id' })
  userId: number;

  @ManyToOne(() => UserEntity, (user) => user.organizationUserRoles)
  @JoinColumn({ name: 'user_id' })
  user: UserEntity;

  @Column('integer', { name: 'organization_id' })
  @Type(() => Number)
  organizationId: number;

  @Column('boolean', { name: 'is_active' })
  isActive: boolean;
}
